### ISSUE TITLE
License of jpeg_memory_src.(cpp|h)

### ISSUE THREAD

**Issue by coldtobi** [dhewm/dhewm3#110]
The both files require to distribute a README file, which is not distributed.
It seems that the text in question is in README.md, but I'm not sure if this fulfilles the requirement, as they say that the file must be named README.  (IANAL)

A workaround I'll use for the Debian package is to require jpeg_mem_src in the jpeg libary, which allows easily to remove the two files in question.

Would that be an option for you too? 

--
tobi

**Comment by smcv** [comment]
Quake III Arena also needs the JPEG memory source, so ioquake3's history might be relevant here. Q3A originally contained a bundled copy of IJG libjpeg which had been modified to add that functionality, but ioquake3 has been changed to have a bundled copy of IJG libjpeg 9 instead, so that adding the separate memory-source files is unnecessary.

jpeg_mem_src() requires either IJG libjpeg 8+ (or 7 might even be enough), or libjpeg-turbo with the jpeg_mem_src API enabled (but in practice, everyone enables that feature).

I think Debian 7 was the last major distribution to not be using libjpeg-turbo as its libjpeg implementation, but its libjpeg was IJG libjpeg 8, which also has jpeg_mem_src(). Debian 8 will use libjpeg-turbo.

tl;dr: I would advise just dropping those files and requiring a vaguely recent libjpeg (either IJG or -turbo will do).

**Comment by coldtobi** [comment]
Yes true, there was a CTTE decission in Debian in favour for libjpeg-turbo, and it will be the default in Jessie

Yes, this is what I do for Debian: Drop those files and require jpeg_mem_src().
Just wondering if it would be safe to generally require a modern libjpeg on common platforms?

The patch I use for Debian is here:
https://alioth.debian.org/plugins/scmgit/cgi-bin/gitweb.cgi?p=pkg-games/dhewm3.git;a=blob;f=debian/patches/02-remove-jpeg_memory_src.patch;h=d8c908afcea10510610cba7b13715d2315a35bbd;hb=HEAD

**Comment by DanielGibson** [comment]
I agree that those files should be removed.

IIRC we introduced them so the game would work with libjpeg62 from debian squeeze, probably before wheezy became stable
